utils/searchFacilityQueryUtils.ts

```diff
@@ -63,70 +63,11 @@ Promise<{ query?: Query<DocumentData>, list?: gqlTypes.Facility[], hasErrors: bo
-                        const fieldName = orderCriterion.fieldToOrder as keyof gqlTypes.Facility
-                        const valueA = facilityA[fieldName]
-                        const valueB = facilityB[fieldName]
-                        let currentComparison = 0
-
-                        if (valueA === undefined || valueA === null) {
-                            if (valueB === undefined || valueB === null) {
-                                // Both are null/undefined, consider them equal
-                                currentComparison = 0
-                            } else {
-                                // A is null/undefined, B is not, A comes first
-                                currentComparison = -1
-                            }
-                        } else if (valueB === undefined || valueB === null) {
-                            // A is not null/undefined, B is, B comes first
-                            currentComparison = 1
-                        } else {
-                            const isValueAComparable = typeof valueA === 'string' || typeof valueA === 'number' || typeof valueA === 'boolean'
-                            const isValueBComparable = typeof valueB === 'string' || typeof valueB === 'number' || typeof valueB === 'boolean'
-
-                            if (isValueAComparable && isValueBComparable) {
-                                currentComparison = comparePrimitiveValues(valueA as ComparablePrimitive, valueB as ComparablePrimitive)
-                            } else {
-                                throw new Error(`Sorting by field '${String(fieldName)}' is not supported. It contains a non-comparable type (e.g., object or array).`)
-                            }
-                        }
-
-                        // Reverse the comparison if the order direction is Descending
-                        if (orderCriterion.orderDirection === gqlTypes.OrderDirection.Desc) {
-                            currentComparison *= -1
-                        }
-
-                        // If a comparison yields a non-zero result, return it immediately.
-                        if (currentComparison !== 0) {
-                            return currentComparison
-                        }
-                    }
-                    // If all criteria are equal, the facilities are considered equal.
-                    return 0
-                })
-            } else {
-                // Default in-memory sorting: sort by 'createdDate' in descending order.
-                allGqlFacilities.sort((facilityA, facilityB) => {
-                    const createdDateA = new Date(facilityA.createdDate)
-                    const createdDateB = new Date(facilityB.createdDate)
-                    return createdDateB.getTime() - createdDateA.getTime()
-                })
-            }
+            const orderCriteria = (filters.orderBy && Array.isArray(filters.orderBy) && filters.orderBy.length > 0)
```

Comment:
can simplify length check. Also, not sure we need to check if it's an array
```suggestion
            const orderCriteria = (filters.orderBy && Array.isArray(filters.orderBy) && filters.orderBy.length)
```